runtime/src/chpl-visual-debug.c

```diff
@@ -64,6 +64,9 @@ static void cb_comm_executeOn_fast(const chpl_comm_cb_info_t *info);
 int chpl_vdebug_fd = -1;
 int chpl_vdebug = 0;
 
+static char buff_tid[CHPL_TASK_ID_STRING_MAX_LEN];
```

Comment:
This can't be static.  If multiple Chapel tasks do TID_STRING() invocations at once chaos will ensue due to the buffer sharing.  The buffer needs to be function-local, and TID_STRING changed to accept both the buffer and the tid as arguments.